File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03309/p03309_num1_parsed.py
# Start of the program

# Read the input N
N = int(input())

# Read the list of integers A
A = list(map(int, input().split()))

# Find the minimum possible sadness of Snuke
# Initialize the min_sadness with a large value
min_sadness = float('inf')

# Iterate through all possible values of b
for b in range(1, N+1):
    # Calculate the sadness of Snuke for the current value of b
    sadness = sum([abs(A[i] - (b+i)) for i in range(N)])
    
    # Update the minimum possible sadness if necessary
    min_sadness = min(min_sadness, sadness)

# Print the minimum possible sadness of Snuke
print(min_sadness)

# End of the program

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03309/p03309_num1_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>

using namespace std;

int main() {
    int N;
    cin >> N;
    
    vector<int> A(N);
    for(int i=0; i<N; i++){
        cin >> A[i];
    }
    
    sort(A.begin(), A.end());
    
    int minSadness = 1e9;
    
    for(int b=A[0]; b<=A[N-1]; b++){
        int sadness = 0;
        
        for(int i=0; i<N; i++){
            sadness += abs(A[i] - (b+i+1));
        }
        
        minSadness = min(minSadness, sadness);
    }
    
    cout << minSadness << endl;
    
    return 0;
}